### PR TITLE
Allow CI to run on branches with this name pattern: feature*

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Allow CI to run on branches with this name pattern: feature*. For example, the netcore3 working branch can be named as `feature-netcore3`

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Comment by iSazonov** [comment]
Will the team collect artifacts for feature branches too? Will there name conflits?

File: .vsts-ci/windows-daily.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ trigger:
     include:
     - master
     - release*
+    - feature*
```
Comment (adityapatwardhan) [inline .vsts-ci/windows-daily.yml]: Do we want daily builds for feature branches right now? The CI builds already run all the tests, should we start daily builds once the branch is more stable?

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: .vsts-ci/windows-daily.yml (hunk 1 above)
Comment (SteveL-MSFT) [reply .vsts-ci/windows-daily.yml]: We probably don't need daily builds on feature branches

Comment (daxian-dbw) [reply .vsts-ci/windows-daily.yml]: Will revert this one.

Comment (daxian-dbw) [reply .vsts-ci/windows-daily.yml]: @adityapatwardhan I still keep `feature*` in the `pr:` section of this file, does that look right?

Comment (adityapatwardhan) [reply .vsts-ci/windows-daily.yml]: we do not use this yml for CI builds. we can remove it from `pr` section too to keep it consistent. when we have CI passing, i guess we will merget his branch to master.

Comment (daxian-dbw) [reply .vsts-ci/windows-daily.yml]: Sounds good. Will revert.

Comment (daxian-dbw) [reply .vsts-ci/windows-daily.yml]: @adityapatwardhan Done. Please take another look. Thanks!

**Comment by adityapatwardhan** [comment]
@daxian-dbw  Just noticed one thing. Should this change be in the `feature` branch or we want to generally enable this? If we make this change in `feature` branch we can drop the change when merging over to master.

**Comment by daxian-dbw** [comment]
Per my discussion with @TravisEz13, we want to generally enable this, so the next time we need to feature branch to work on, the CI will be enabled automatically.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
I will wait for th CI to finish then merge this. I will also make the branch a protected branch and make the CI 'required'

**Comment by daxian-dbw** [comment]
Sounds good. Thanks @adityapatwardhan

**Pull request merged**